Solutions/Azure Active Directory/Analytic Rules/NRT_UseraddedtoPrivilgedGroups.yaml

```diff
@@ -25,36 +25,47 @@ query: |
-  | mv-expand bagexpansion=array modProps
-  | evaluate bag_unpack(modProps)
-  | extend displayName = column_ifexists("displayName", "NotAvailable"), newValue = column_ifexists("newValue", "NotAvailable")
-  | where displayName =~ "Role.WellKnownObjectName"
-  | extend DisplayName = displayName, GroupName = replace('"','',newValue)
-  | extend initByApp = parse_json(InitiatedBy).app, initByUser = parse_json(InitiatedBy).user
-  | extend AppId = initByApp.appId,
-  InitiatedByDisplayName = case(isnotempty(initByApp.displayName), initByApp.displayName, isnotempty(initByUser.displayName), initByUser.displayName, "not available"),
-  ServicePrincipalId = tostring(initByApp.servicePrincipalId),
-  ServicePrincipalName = tostring(initByApp.servicePrincipalName),
-  UserId = initByUser.id,
-  UserIPAddress = initByUser.ipAddress,
-  UserRoles = initByUser.roles,
-  UserPrincipalName = tostring(initByUser.userPrincipalName),
-  TargetUserPrincipalName = tostring(TargetResources.userPrincipalName)
+  | mv-apply TargetResource = TargetResources on 
+    (
+        where TargetResource.type =~ "User"
+        | extend TargetUserPrincipalName = tostring(TargetResource.userPrincipalName),
+                 modProps = TargetResource.modifiedProperties
+    )
+  | mv-apply Property = modProps on 
+    (
+        where Property.displayName =~ "Role.WellKnownObjectName"
+        | extend DisplayName = trim('"',tostring(Property.displayName)),
+                 GroupName = trim('"',tostring(Property.newValue))
+    )
+  | extend AppId = InitiatedBy.app.appId,
+        InitiatedByDisplayName = case(isnotempty(InitiatedBy.app.displayName), InitiatedBy.app.displayName, isnotempty(InitiatedBy.user.displayName), InitiatedBy.user.displayName, "not available"),
+        ServicePrincipalId = tostring(InitiatedBy.app.servicePrincipalId),
+        ServicePrincipalName = tostring(InitiatedBy.app.servicePrincipalName),
+        UserId = InitiatedBy.user.id,
+        UserIPAddress = InitiatedBy.user.ipAddress,
+        UserRoles = InitiatedBy.user.roles,
+        UserPrincipalName = tostring(InitiatedBy.user.userPrincipalName)
   | where GroupName in~ (PrivilegedGroups)
   // If you don't want to alert for operations from PIM, remove below filtering for MS-PIM.
   //| where InitiatedByDisplayName != "MS-PIM"
   | project TimeGenerated, AADOperationType, Category, OperationName, AADTenantId, AppId, InitiatedByDisplayName, ServicePrincipalId, ServicePrincipalName, DisplayName, GroupName, UserId, UserIPAddress, UserRoles, UserPrincipalName, TargetUserPrincipalName
-  | extend AccountCustomEntity = case(isnotempty(ServicePrincipalName), ServicePrincipalName, isnotempty(ServicePrincipalId), ServicePrincipalId, isnotempty(UserPrincipalName), UserPrincipalName, "not available")
+  | extend AccountCustomEntity = case(isnotempty(ServicePrincipalName), ServicePrincipalName, 
+                                      isnotempty(UserPrincipalName), UserPrincipalName,  
```

Comment:
@petebryan I have removed the ServicePrincipleId from AccountCustomEntity because that is not a valid Account Name. Should we add this back in some other way?